pkg/controller/releasemanager/incarnation.go

```diff
@@ -296,6 +338,18 @@ func (i *Incarnation) isReleaseEligible() bool {
 	return i.status.ReleaseEligible
 }
 
+// Pre-exit hook to execute code just before exiting a state
+func (i *Incarnation) exitState(ctx context.Context, state string) error {
```

Comment:
1. I don't think this will work because of the bug I ran into this morning, the one where I need to call retire() from everywhere.
2. why not make exit() a function of the state?

Reply:
Removed the exit hook completely.